code/xcvm/evm/src/Gateway.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
```

Comment:
In Cosmwasm contracts, we use `Router` to handle `Gateway`'s and `Router`'s responsibilities. I think we should settle on whether we use `Gateway` or `Router`.

Reply:
We might want to merge the router and gateway contract as they are global singletons anyway?